apis/api/commontypes/astarte_common_types.go

```diff
@@ -344,6 +344,14 @@ type AstarteVerneMQSpec struct {
 	PersistentClientExpiration string `json:"persistentClientExpiration,omitempty"`
 	// +optional
 	MirrorQueue string `json:"mirrorQueue,omitempty"`
+	// This option allows, when true, to handle SSL termination at VerneMQ level.
+	// Default: false
+	// +optional
+	SSLListener *bool `json:"sslListener,omitempty"`
+	// Reference to the secret containing the TLS certificate for VerneMQ.
+	// The field will be used only if SSLListener is set to true.
+	// +optional
+	SSLListenerCert *v1.SecretReference `json:"sslListenerCert,omitempty"`
```

Comment:
Turns out that by the way you're using it, specifying the namespace simply wouldn't work. I suggest then changing this to a `string` rather than a ref, and the name should be something along the lines of `SSLListenerCertSecretName`. Pods won't support mounting secrets outside of their namespace (see https://v1-18.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#secretvolumesource-v1-core)